CONTRIBUTING.md

```diff
@@ -76,11 +76,14 @@ Maintainers are:
 ### PR Checklist
 
 Before your PR can be merged it must meet the following criteria:
+* Informative PR Description alongside an overview of the changes and a high-level design in case of a new feature
+* Commits split logically with a clear purpose.
+* Commits title concise and clear while commit message informative stating the nature and purpose of the changes in the commit.
```

Comment:
* Currently our merge policy is squashing all PR commits, so consider:
1. Splitting to incremental small PRs instead of a full feature PR
1. Interactive rebase (git rebase -i) before PR is getting merged, so it won’t look like a list of useless titles